Adapted Java quickstart to structure of Python quickstart (#153)
emnify/product-docs: docs/sdks/java/quickstart.md

# Getting started with the emnify Java SDK

## Installation

The emnify Java SDK is open source and the source code is available on [GitHub](https://github.com/EMnify/emnify-sdk-java).

To install the SDK, add the emnify dependency to your existing project and use build automation tools like Maven or Gradle or install it yourself manually.

:::caution
Regardless of the package manager you are using, you need to specify the latest version of the emnify Java SDK.
Please use `{version}` to do so, as demonstrated in the following example. :::

### Prerequisites

> <b>Supported Java versions:</b>

Reviewer: We should use [Markdown formatting](https://www.markdownguide.org/basic-syntax/) wherever possible (this might also be why this section isn't rendering properly, see screenshot below).

If you want to keep it bold:

```suggestion
**Supported Java versions:**
```

Alternatively, we could turn this into an h4 (my preference):

```suggestion
#### Supported Java versions
```

<details><summary>Screenshot of Prerequisites section on current branch</summary>
<img width="730" alt="Screenshot 2023-08-02 at 12 56 14" src="https://github.com/emnify/product-docs/assets/26869552/c8aa4f9e-d880-4e43-9571-74972fdb6faf">
</details>

[OpenJDK 8](https://openjdk.org/projects/jdk8/), [OpenJDK 11](https://openjdk.org/projects/jdk/11/), [OracleJDK 8](https://www.oracle.com/java/technologies/downloads/#java8) and [OracleJDK 11](https://www.oracle.com/java/technologies/downloads/#java11).

> <!-- Alternative

Reviewer: This would be my choice, as lists tend to be easier for readers to scan and comprehend quickly. What do you think?

- [OpenJDK 8](https://openjdk.org/projects/jdk8/)
- [OpenJDK 11](https://openjdk.org/projects/jdk/11/)
- [OracleJDK 8](https://www.oracle.com/java/technologies/downloads/#java8)
- [OracleJDK 11](https://www.oracle.com/java/technologies/downloads/#java11)
-->

> <b>TLS requirements: </b>

Reviewer: Same note as for "Supported Java versions" 😁

# Getting started with the emnify Java SDK

## Installation

The emnify Java SDK is open source and the [source code is available on GitHub](https://github.com/EMnify/emnify-sdk-java).

To install the SDK, add the emnify dependency to your existing project and use build automation tools, like Maven or Gradle, or install it manually.

:::caution
Regardless of the package manager you are using, you need to specify the latest version of the emnify Java SDK.
The following installation examples use *`SDK_VERSION`* where this should be specified.
:::

### Prerequisites

> #### Supported Java versions:

Reviewer: Since we're going with headers, we don't need the colons.

```suggestion
#### Supported Java versions
```

> [OpenJDK 8](https://openjdk.org/projects/jdk8/), [OpenJDK 11](https://openjdk.org/projects/jdk/11/), [OracleJDK 8](https://www.oracle.com/java/technologies/downloads/#java8) and [OracleJDK 11](https://www.oracle.com/java/technologies/downloads/#java11).

Reviewer: ```suggestion

- [OpenJDK 8](https://openjdk.org/projects/jdk8/)
- [OpenJDK 11](https://openjdk.org/projects/jdk/11/)
- [OracleJDK 8](https://www.oracle.com/java/technologies/downloads/#java8)
- [OracleJDK 11](https://www.oracle.com/java/technologies/downloads/#java11)
```

Unless you have an argument against it, I'd prefer this stay in a list format 👩🏼‍🎨

> #### TLS requirements:

Reviewer: ```suggestion
#### TLS requirements

```

> Replace *`VERSION_NUMBER`* with the version of Maven you're using.

Reviewer: ```suggestion
Replace *`SDK_VERSION`* with the version of the emnify Java SDK you're using.
```

My mistake (see earlier comment) 🤪 If you could also update the placeholder in the code to say SDK_VERSION instead of VERSION_NUMBER, that'd be amazing 🙏🏼

### Install with Gradle

Use the following to add the emnify dependency to your project via [Gradle](https://gradle.org/install/):
```gradle
    implementation group: "com.emnify.sdk", name: "emnify", version: "VERSION_NUMBER"
```

> Replace *`VERSION_NUMBER`* with the version of Gradle you're using.

Reviewer: ```suggestion
Replace *`SDK_VERSION`* with the version of Gradle you're using.
```

Same comment and request as the Maven section 😇

  mvn package
```

## Configuration

### Environment variables

> To store configuration settings like an application token or a base URL, use environment variables instead of hardcoding them in your application. Here are some examples:

Reviewer: Why'd you decide reword this sentence? Asking because I think this wording is less clear than the original, but I might be missing context 🤔 

Either way, we want to keep each sentence in Markdown files on it's own line. That way, we can better see/track the diff.

```suggestion
To store configuration settings, like an application token or a base URL, use environment variables instead of hardcoding them in your application. 
Here are some examples:
```

Author: Ouh I didn't know that "one sentence - one line" thing. Good to know, thanks!

| Name               | Description                                                              |
|--------------------|--------------------------------------------------------------------------|
| `EMNIFY_BASE_PATH` | Base URL to form a request. Default value:  `https://cdn.emnify.net` |
| `EMNIFY_APPLICATION_TOKEN`  | Variable for authenticating via an [application token](/rest/authentication#authenticate-with-an-application-token). |
| `EMNIFY_USERNAME` and `EMNIFY_PASSWORD` | Variables for authenticating via [username and password](/rest/authentication#authenticate-with-user-credentials). |

## Explore more 

> If you're new to IoT connectivity and emnify, start by learning some [common terminology and concepts](/sdks/concepts) and explore some [use cases](/sdks/java/examples) that show what the SDK is capable of. 

Reviewer: This is a big improvement, thank you 🙌🏼

Author: Thanks!!
